### DISCUSSION TITLE
aggiungere valore numerico ad un contatore numerico ...

### DISCUSSION THREAD

**Discussion by RobyBS** [indomus/forum#1676] [HUB: Home Assistant]
scusate .. ma non ne vengo fuori ... e sicuramente l'errore è banale o di sintassi ... non trovo esempi ... 

ho persino fatto una cosa simile tra due contatori ( sensor ) e la cosa è andata  a buon fine ma nel caso sotto HA mi riscontra errore ...

Ho uno shelly EM che uso per contabilizzare il consumo ( prelievo ) dalla rete elettrica di casa quando ricarico la mia vettura elettrica .
Lo shelly EM lo ho inserito successivamente ( da 3 giorni ) .... prima utilizzavo un semplice contabilizzatore elettronico che appunto fino a 3 giorni fa ha contabilizzato un certo valore ( per la precisione 1.266,23 watt ) 
Adesso voglio creare un sensore dato dalla somma della contabilizzazione dello shelly EM + questi 1.266 watt che visualizzerò in HA per il futuro ... 

questa la mia configurazione :

```
  - platform: template
    sensors:
      shelly_em_totale_tesla:
        friendly_name: "Shelly EM Tesla Totale"
        value_template: '{{ (states("sensor.auto_tesla_m3_energy") | float + 0.1266 }}'
        unit_of_measurement : "Watt"
```

questo l'errore che mi restituisce HA prima del riavvio del sistema : 

`[547937914448] The system cannot restart because the configuration is not valid: Invalid config for [sensor.template]: invalid template (TemplateSyntaxError: unexpected '}', expected ')') for dictionary value @ data['sensors']['shelly_em_totale_tesla']['value_template']. Got '{{ (states("sensor.auto_tesla_m3_energy") | float + 0.1266 }}'. (See ?, line ?).`

grazie per l'aiuto 👍

**Comment by snakuzzo** [comment]
manca una parentesi...vedi bene

```yaml
value_template: "{{ ( states('sensor.auto_tesla_m3_energy') | float ) + 0.1266 }}"
```

**Reply by RobyBS** [reply]
perfetto ... tutto funziona .. solo che l'unità da aggiungere è 1266  e non 0.1266 per esprimere 1266 watt ... 👍 👍 👍

**Reply by snakuzzo** [reply]
quello non l'ho nemmeno letto :) ...sono semplicemente intervenuto sulla sintassi

**Comment by RobyBS** [comment]
rispolvero la discussione .. in quanto mi succede una cosa strana ... 

il sensor creato sopra che nel frattempo ho rinominato in:
```
- platform: template
    sensors:
      shelly_em_tesla_totale:
        friendly_name: "Shelly EM Tesla Totale"
        value_template: "{{ ( states('sensor.auto_tesla_m3_energy') | float ) + 1266 }}"
        unit_of_measurement: "Watt"
```

quando lo aggiungo alla dashboard in una scheda , mi capita quanto segue : 
- se lo aggiungo selezionandolo in una scheda semplice , mi appare il suo valore 
```
type: entities
entities:
  - sensor.shelly_em_tesla_totale
```
- se lo aggiungo invece in una scheda un po' più complessa tipo questa sotto , mi dice che **non ha ID univoco** e non può essere gestita :
```
chart_type: line
period: day
days_to_show: 360
type: statistics-graph
entities:
  - sensor.shelly_em_tesla_totale
stat_types:
  - sum
title: Tesla Consumo Tot.
```

ma se non ha ID univoco non può essere gestita nè prima nè dopo , o sbaglio ? 
ho dato uno sguardo su come risolvere il conflitto , ma non riesco a capire come fare , anche perchè non tutte le opzioni sono disponibili nel mio caso rispetto a quello dell'esempio di questa pagina purtroppo ... ( https://www.home-assistant.io/docs/configuration/customizing-devices ) ...

**Reply by RobyBS** [reply]
rispolvero per sapere se qualcuno ha una soluzione .. 
grazie
👍
